### PR TITLE
feat: add regex username and password feature

### PR DESCRIPTION
**Pull request by BaraKharseh**
Implemented the feature that does not allow the creation of user accounts without follow the regex format we have required for usernames and password. In order to create a user account, the username must contain 6 characters and be completely alphanumeric (no special characters or spaces) and the password must contain 8 characters which can be anything. Also, if a user account does not get created, the reason is provided: either the format does not follow the regex or the username is already taken.

### CODE REVIEW

File: src/controllers/SignInController.java [hunk 1]
```diff
@@ -53,32 +54,58 @@ public LinkedHashMap<String, Command> AllCommands() {
...
+            return true;
+        }
+        return false;
+    }
+
+    private void signIn(String username, String password) {
+        boolean signedIn = adminSignIn(username, password) |
+                            patientSignIn(username, password) |
+                            doctorSignIn(username, password) |
+                            secretarySignIn(username, password);
```
Comment (ahmedkhalf) [inline src/controllers/SignInController.java:105]: Please use logical or instead of bitwise or: `||`.

File: src/useCases/AdminManager.java [hunk 2]
```diff
@@ -27,13 +29,18 @@ public AdminManager(Database database){
      * @param password password of new account.
      * @return Admin data if sign in successful, if username exists in the database, return null.
      */
-    public AdminData createAdmin(String username, String password){
-        Admin admin = new Admin(username, password);
-        if (adminDatabase.add(admin) != null) {
-            admin.setContactInfoId(newContactInDatabase());
-            return new AdminData(admin);
+    public AdminData createAdmin(String username, String password) {
+        if (Pattern.matches("^[a-zA-Z0-9]{6,}$", username) && Pattern.matches("^.{8,}$", password)) {
```
Comment (ahmedkhalf) [inline src/useCases/AdminManager.java]: It looks like you are repeating the same pattern across all UserManagers. Creating a helper method inside the UserManager abstract class that takes arguments `username` and `password` and returns a boolean would do you better.

**Review by ahmedkhalf — CHANGES_REQUESTED** [review]
Well done.

**Review by ahmedkhalf — APPROVED** [review]
Ready for merge.

**Review by ryanMMM — APPROVED** [review]
LGTM!!

**Pull request merged**